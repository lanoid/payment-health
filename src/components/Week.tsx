import * as React from 'react';
import Day from './Day';

interface Props {
    week: any[];
}

export default class Week extends React.Component<Props> {
    static defaultProps: Props = {
        week: []
    }
    render() {
        return (
            <ul className={`week`}>
                {this.props.week.map((day, i) => <Day key={i} day={day} />)}
            </ul>
        )
    }
}