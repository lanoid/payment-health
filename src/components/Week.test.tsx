import * as React from 'react';
import { shallow } from 'enzyme';
import Week from './Week';

describe('Week', () => {
    const week = shallow(<Week />);

    it('renders a single week ul', () => {
        expect(week.find('ul').length).toBe(1);
        expect(week.find('.week').length).toBe(1);
    });
    
    it('matches snapshot', () => {
        expect(week).toMatchSnapshot();
    });
});