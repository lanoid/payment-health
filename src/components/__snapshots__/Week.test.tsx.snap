// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week matches snapshot 1`] = `
<ul
  className="week"
/>
`;
